channeldb/edge_policy.go

```diff
@@ -39,26 +39,89 @@ const (
 	edgePolicy2EncodingType edgePolicyEncodingType = 0
 )
 
+type edgePolicyEncodingInfo struct {
+	updateBucketKey []byte
+	updateKey       []byte
+	serialize       func(w io.Writer, toNode []byte) error
+	typeByte        func() (edgePolicyEncodingType, bool)
+}
+
+func encodingInfoFromEdgePolicy(policy models.ChannelEdgePolicy) (
+	*edgePolicyEncodingInfo, error) {
+
+	switch p := policy.(type) {
+	case *models.ChannelEdgePolicy1:
+		updateUnix := uint64(p.LastUpdate.Unix())
+		var indexKey [8 + 8]byte
+		byteOrder.PutUint64(indexKey[:8], updateUnix)
+		byteOrder.PutUint64(indexKey[8:], p.ChannelID)
+
+		return &edgePolicyEncodingInfo{
+			updateBucketKey: edgeUpdateIndexBucket,
+			updateKey:       indexKey[:],
+			serialize: func(w io.Writer, toNode []byte) error {
+				copy(p.ToNode[:], toNode)
+
+				return serializeChanEdgePolicy1(w, p)
+			},
+			typeByte: func() (edgePolicyEncodingType, bool) {
+				return 0, false
+			},
+		}, nil
+
+	case *models.ChannelEdgePolicy2:
+		indexKey := make([]byte, 4+8)
```

Comment:
Why do we need the block height _and_ the scid value? The scid encodes a block height (unrolled)